protocol/binary/stream_reader_test.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package binary
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func BenchmarkReadString(b *testing.B) {
+	var streamBuff bytes.Buffer
+
+	// Encode with Streaming protocol
+	w := NewStreamWriter(&streamBuff)
+	require.NoError(b, w.WriteString("the quick brown fox jumps over the lazy dog"), "WriteString")
+	require.NoError(b, w.Close(), "StreamWriter Close")
```

Comment:
nit: Since the content of the error strings are the names of the methods getting called I think we can drop this.

Reply:
Sure, I can do that. But could you please help me understand this better?

I assumed that the optional message helps developers quickly understand the test failure without having to browse the test code. Is that correct? 

If we drop the message from the require.NoError() call, it will print the following on test failure:
```
--- FAIL: BenchmarkReadString
    stream_reader_test.go:37: 
                Error Trace:    stream_reader_test.go:37
                Error:          Received unexpected error: <>.
                Test:           BenchmarkReadString
```
It doesn't print the name of the method getting called. The only clue for the developer is the line number where the assertion failed.